code/src/components/product-card/ProductCard.test.tsx

```diff
@@ -19,78 +20,77 @@ const mockProduct: Product = {
-}));
+jest.mock("@/hooks/use-add-to-cart-or-open-drawer/useAddToCartOrOpenDrawer");
 
-describe("ProductCard component", () => {
-  beforeEach(() => {
-    renderWithProviders(
-      <ProductCard
-        product={mockProduct}
-        onCartIconClick={mockCartIconClickHandler}
-        isUserAuthorized
-        isInCart
-      />
-    );
+const renderAndMock = (isProductInCart: boolean) => {
+  (useAddToCartOrOpenDrawer as jest.Mock).mockReturnValue({
+    isProductInCart,
+    addToCartOrOpenDrawer: mockAddToCartOrOpenDrawer
   });
+  renderWithProviders(<ProductCard product={mockProduct} />);
+};
 
+describe("ProductCard component", () => {
   afterEach(() => {
     jest.clearAllMocks();
   });
 
-  test("should render product name", () => {
-    const productName = screen.getByText(mockProduct.name);
-    expect(productName).toBeInTheDocument();
-  });
-
-  test("should render product price", () => {
-    const productPrice = screen.getByText(formatPrice(mockProduct.price));
-    expect(productPrice).toBeInTheDocument();
-  });
-
-  test("should render product description", () => {
-    const productDescription = screen.getByText(mockProduct.description);
-    expect(productDescription).toBeInTheDocument();
-  });
+  describe("when product is not in cart", () => {
+    beforeEach(() => {
+      renderAndMock(false);
+    });
 
-  test("should render product image with correct src and alt attributes", () => {
-    const productImage = screen.getByRole("img", { name: mockProduct.name });
-    expect(productImage).toHaveAttribute("src", mockProduct.image);
-    expect(productImage).toHaveAttribute("alt", mockProduct.name);
-  });
+    test("should render product name", () => {
+      const productName = screen.getByText(mockProduct.name);
+      expect(productName).toBeInTheDocument();
+    });
 
-  test("should render product link", () => {
-    const productLink = screen.getByRole("link");
-    expect(productLink).toHaveAttribute(
-      "href",
-      routes.productDetails.path(mockProduct.id)
-    );
-  });
+    test("should render product price", () => {
+      const productPrice = screen.getByText(formatPrice(mockProduct.price));
+      expect(productPrice).toBeInTheDocument();
+    });
 
-  test("should call useState on mount", () => {
-    expect(mockSetState).toHaveBeenCalledWith(true);
-  });
+    test("should render product description", () => {
+      const productDescription = screen.getByText(mockProduct.description);
+      expect(productDescription).toBeInTheDocument();
+    });
 
-  test("Should call functions on cart icon click", () => {
-    const cartIcon = screen.getByTestId("add-to-cart-button");
+    test("should render product image with correct src and alt attributes", () => {
+      const productImage = screen.getByRole("img", { name: mockProduct.name });
+      expect(productImage).toHaveAttribute("src", mockProduct.image);
+      expect(productImage).toHaveAttribute("alt", mockProduct.name);
```

Comment:
`src` will be enough I guess. why do you add `alt` here ?

Reply:
I removed it